### ISSUE TITLE
No way to know if parameter is set with --no- options

### ISSUE THREAD

**Issue by eisbehr-** [tj/commander.js#630]
Hello,

I want to use two parameters on a command, somehow like this:
```JS
program.option("-d, --debug", "overwrite debug configuration with 'true'")
       .option("-n, --no-debug", "overwrite debug configuration with 'false'");
```

My program has a configuration file, where a default value for `debug` is defined. The parameters `-d` and `-n` should override this setting, but only when set. If not set the default value should be picked. So for example:

Default value for `debug` in configuration is `true`:
```TETX
command      <-  debug is enabled
command -d   <-  debug is enabled
command -n   <-  debug is disabled
```

Default value for `debug` in configuration is `false`:
```TETX
command      <-  debug is disabled
command -d   <-  debug is enabled
command -n   <-  debug is disabled
```

The problem is, when using a `--no-` parameter, `commander` always set a default value for `debug` and there is no way to see, if one of those options was set while execute the command. I would need a way to see if `-d` or `-n` was really given by the user.

Otherwise, it will always overwrite my configuration and not only when the user want to do it especially.

**Comment by usmonster** [comment]
Hey, I may have solved this in #795. Please try it out and let me know?

**Comment by shadowspawn** [comment]
Closing in favour of #108

**Comment by shadowspawn** [comment]
This issue will be resolved when v3.0.0 is released. Available now as a prerelease. See #1001
